code/modules/uplink/uplink_items/dangerous.dm

```diff
@@ -0,0 +1,101 @@
+	name = "Toy Pistol with Riot Darts"
+	desc = "An innocent-looking toy pistol designed to fire foam darts. Comes loaded with riot-grade \
+			darts effective at incapacitating a target."
+	item = /obj/item/gun/ballistic/automatic/pistol/toy/riot
+	cost = 2
+	surplus = 10
+
+// Low progression cost
+
+/datum/uplink_item/dangerous/throwingweapons
+	name = "Box of Throwing Weapons"
+	desc = "A box of shurikens and reinforced bolas from ancient Earth martial arts. They are highly effective \
+			throwing weapons. The bolas can knock a target down and the shurikens will embed into limbs."
+	progression_minimum = 10 MINUTES
+	item = /obj/item/storage/box/syndie_kit/throwing_weapons
+	cost = 3
+	illegal_tech = FALSE
+
+/datum/uplink_item/dangerous/sword
+	name = "Energy Sword"
+	desc = "The energy sword is an edged weapon with a blade of pure energy. The sword is small enough to be \
+			pocketed when inactive. Activating it produces a loud, distinctive noise."
+	progression_minimum = 20 MINUTES
+	item = /obj/item/melee/energy/sword/saber
+	cost = 8
+	purchasable_from = ~UPLINK_CLOWN_OPS
+
+/datum/uplink_item/dangerous/powerfist
+	name = "Power Fist"
+	desc = "The power-fist is a metal gauntlet with a built-in piston-ram powered by an external gas supply.\
+			Upon hitting a target, the piston-ram will extend forward to make contact for some serious damage. \
+			Using a wrench on the piston valve will allow you to tweak the amount of gas used per punch to \
+			deal extra damage and hit targets further. Use a screwdriver to take out any attached tanks."
+	progression_minimum = 20 MINUTES
+	item = /obj/item/melee/powerfist
+	cost = 6
+
+/datum/uplink_item/dangerous/rapid
+	name = "Gloves of the North Star"
+	desc = "These gloves let the user punch people very fast. Does not improve weapon attack speed or the meaty fists of a hulk."
+	progression_minimum = 20 MINUTES
+	item = /obj/item/clothing/gloves/rapid
+	cost = 8
+
+
+// Medium progression cost
+
+/datum/uplink_item/dangerous/doublesword
+	name = "Double-Bladed Energy Sword"
+	desc = "The double-bladed energy sword does slightly more damage than a standard energy sword and will deflect \
+			all energy projectiles, but requires two hands to wield."
+	progression_minimum = 30 MINUTES
+	item = /obj/item/dualsaber
+
+	cost = 16
+	purchasable_from = ~UPLINK_CLOWN_OPS
+
+/datum/uplink_item/dangerous/doublesword/get_discount()
+	return pick(4;0.8,2;0.65,1;0.5)
+
+/datum/uplink_item/dangerous/guardian
+	name = "Holoparasites"
+	desc = "Though capable of near sorcerous feats via use of hardlight holograms and nanomachines, they require an \
+			organic host as a home base and source of fuel. Holoparasites come in various types and share damage with their host."
+	progression_minimum = 30 MINUTES
+	item = /obj/item/storage/box/syndie_kit/guardian
+	cost = 18
+	surplus = 0
+	purchasable_from = ~(UPLINK_NUKE_OPS | UPLINK_CLOWN_OPS)
+	restricted = TRUE
+
+/datum/uplink_item/dangerous/pistol
```

Comment:
I'm not sure what the point of the Makarov is if the revolver is available at the same value. Besides being cheaper, at this point in the round you want either to expediently kill someone or you have to expect a 10 minute blaze of glory with a makarov + mags, and even that is still better accomplished by just using a revolver + speedloader.

The makarov was strongest when used conservatively, emphasizing long-term planning and repeat skirmishes over multiple conflicts in a round (hence the affordability of the weapon + ammo + portability). Maybe this item needs a re-assessment rather than any particular issue with it coming out at 30 minutes, like bundling it with some spare mags.